coils_tet_mesh.py

```diff
@@ -0,0 +1,189 @@
+                the .step file and will append the index to the .exo filename.**
+            h5m_general_path (str): absolute path to a .h5m file that will be
+                created by the function. 
+                **Note: Do not include any numerical index at the
+                end of the file name because the function iterates over every coil in
+                the .step file and will append the index to the .h5m filename.**
+        """
+
+    # GEOMETRY AND MESH CREATION
+
+    # Start Cubit - this step is key
+    cubit.init(['cubit', '-nojournal'])
+
+    # Import coils
+    cubit.cmd(f'import step "{coils_path}" heal')
+    cubit.cmd('volume  size auto factor 5')
+
+    # For-loop iterates over all 48 coils included in the coils.step file
+    for coil_num in np.arange(1,total_coil_count):
+        
+        # Create the mesh
+        cubit.cmd(f'volume {coil_num} scheme tetmesh')
+        cubit.cmd(f'volume {coil_num} scheme tetmesh')
+        cubit.cmd(f'mesh volume {coil_num}')
+
+        # Set log file path and initiate logging
+        log_split = log_general_path.split('.')
+        log_path = f"{log_split[0]}_{coil_num}.{log_split[1]})"
+        cubit.cmd(f'logging on file "{log_path}"')
+
+        # Mesh quality metrics
+        quality_metric_list = ['shape', 'aspect ratio', 'condition no.', 'distortion', 'element volume',
+                                'equiangle skew', 'equivolume skew','inradius', 'jacobian', 
+                                'normalized inradius', 'relative size', 'scaled jacobian', 'shape and size']
+
+        # Run mesh quality metrics for each relevant metric
+        for metric in quality_metric_list:
+            command = f'quality volume {coil_num} {metric} global draw mesh list'
+            mesh_analytics = cubit.cmd(command)
+
+
+        # MESH QUALITY ANALYSIS
+
+        # Read the text file
+        with open(log_path, 'r') as file:
+            text = file.read()
+
+        # Cut out each of the string 'tables' 
+        start_table = 'Function Name'
+        end_table = 'Finished Command:'
+        ranges = []
+        start_index = 0
+
+        while True:
+            # Find the starting index after the previous occurrence
+            start_index = text.find(start_table, start_index)
+
+            # Break the loop if the starting cue is not found
+            if start_index == -1:
+                break
+
+            # Find the ending index after the starting index
+            end_index = text.find(end_table, start_index + 1)
+
+            # Break the loop if the ending cue is not found
+            if end_index == -1:
+                break
+
+            # Extract the range between the starting and ending index
+            range_text = text[start_index : end_index + 1]
+            ranges.append(range_text)
+
+            # Update the starting index for the next iteration
+            start_index = end_index + 1
```

Comment:
Maybe this can be a function to reduce cognitive load

Reply:
Arguably, the file reading should perhaps be outside the function for separation of concerns